pcdsdevices/mirror.py

```diff
@@ -293,8 +293,42 @@ class HardOffsetMirror(Device, BaseInterface):
     name : ``str``
         Alias for the device
     """
-    not_implemented = Cpt(SignalRO, name="Not Implemented",
-                          value="Not Implemented", kind='normal')
+    # Motor components: can read/write positions
+    y_up = Cpt(EpicsSignal, ':MMS:YUP', name='y_up', kind='normal')
+    y_dwn = Cpt(EpicsSignal, ':MMS:YDWN', name='y_dwn', kind='config')
+    x_up = Cpt(EpicsSignal, ':MMS:XUP', name='x_up', kind='normal')
+    x_dwn = Cpt(EpicsSignal, ':MMS:XDWN', name='x_dwn', kind='config')
+    pitch = Cpt(EpicsSignal, ':MMS:PITCH', name='pitch', kind='normal')
+    bender = Cpt(EpicsSignal, ':MMS:BENDER', name='bender', kind='normal')
+
+    # Gantry components
+    gantry_x = Cpt(EpicsSignalRO, ':GANTRY_X_RBV', name='gantry_x',
+                   kind='normal')
+    gantry_y = Cpt(EpicsSignalRO, ':GANTRY_Y_RBV', name='gantry_y',
+                   kind='normal')
+    couple_y = Cpt(EpicsSignal, ':COUPLE_Y', name='couple_y', kind='config')
+    couple_x = Cpt(EpicsSignal, ':COUPLE_X', name='couple_x', kind='config')
+    decouple_y = Cpt(EpicsSignal, ':DECOUPLE_Y', name='decouple_y',
+                     kind='config')
+    decouple_x = Cpt(EpicsSignal, ':DECOUPLE_X', name='decouple_x',
+                     kind='config')
+    couple_status_y = Cpt(EpicsSignalRO, ':ALREADY_COUPLED_Y_RBV',
+                          name='already_coupled_y', kind='normal')
+    couple_status_x = Cpt(EpicsSignalRO, ':ALREADY_COUPLED_X_RBV',
+                          name='already_coupled_x', kind='normal')
+
+    # RMS Cpts:
+    y_enc_rms = Cpt(EpicsSignalRO, ':Y_ENC:RMS_RBV', name='y_enc_rms',
+                    kind='normal')
+    x_enc_rms = Cpt(EpicsSignalRO, ':X_ENC:RMS_RBV', name='x_enc_rms',
+                    kind='normal')
+    pitch_enc_rms = Cpt(EpicsSignalRO, ':PITCH_ENC:RMS_RBV',
+                        name='pitch_enc_rms', kind='normal')
+    bender_enc_rms = Cpt(EpicsSignalRO, ':BENDER_ENC:RMS_RBV',
+                         name='bender_enc_rms', kind='normal')
+
+    def __init__(self, prefix, **kwargs):
+        super().__init__(prefix, **kwargs)
```

Comment:
You do not need to subclass `__init__` if you aren't changing anything

Reply:
Oh okay so now that I've removed this statement, all the parameters I pass to `XOffsetMirror` essentially go to `Device.__init__`? And `XOffsetMirror` doesn't have its own `__init__` because there aren't any parameters that I want to specifically passed to it?

Reply:
Exactly. If you don't define a particular method, it calls the method of the parent class instead. Here you had overwritten the `__init__` method to simply call the parent's `__init__` method with no functional changes, which is the same as if you didn't overwrite it at all.

Note that there were two non-functional changes you made by overriding the `__init__` method: you hid the parent's `__init__` docstring (if it existed) and you changed the outward-facing signature to have less args inspectable from `IPython`, e.g. with `__init__` redefined as above it's not clear that there is a mandatory `name` `kwarg`

Reply:
Makes sense, thanks its always good to have a refresher on class inheritance peculiarities.